core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1097,8 +1097,8 @@ class DAGScheduler(
             throw new SparkException(s"attempted to access non-existent accumulator $id")
         }
         acc.merge(updates.asInstanceOf[AccumulatorV2[Any, Any]])
-        // To avoid UI cruft, ignore cases where value wasn't updated
-        if (acc.name.isDefined && !updates.isZero) {
+        // Only display named accumulators on UI.
+        if (acc.name.isDefined) {
```

Comment:
it's because the executor no longer sends back updates where the value is zero, so the second condition is always assumed to be true. Maybe we should add an assert or something instead.